test/modules/paymentProcessor/bridging/abstracts/PP_Connext_Crosschain_v1_Test.t.sol

```diff
@@ -0,0 +1,457 @@
+        _orchestrator.executeAddModule(address(paymentClient));
+
+        //Init payment client
+        paymentClient.init(_orchestrator, _METADATA, bytes(""));
+        paymentClient.setIsAuthorized(address(paymentProcessor), true);
+        paymentClient.setToken(token);
+        impl = address(new PP_Connext_Crosschain_v1());
+        crossChainManager = PP_Connext_Crosschain_v1(Clones.clone(impl));
+        // Setup mock addresses
+        mockConnextBridge = address(0x123456);
+        mockEverClearSpoke = address(everclearPaymentMock); // Using the existing mock
+        mockWeth = address(0x789012); // Or deploy a mock WETH contract if needed
+
+        //init the processor
+        // Initialize with proper config data
+        bytes memory configData = abi.encode(
+            address(mockEverClearSpoke), // address of your EverClear spoke contract
+            address(mockWeth) // address of your WETH contract
+        );
+
+        crossChainManager.init(_orchestrator, _METADATA, configData);
+        paymentClient.setIsAuthorized(address(crossChainManager), true);
+
+        // Setup token approvals and initial balances
+        token.mint(address(this), 1000 ether);
+        token.approve(address(crossChainManager), type(uint).max);
+
+        // Add these lines to ensure proper token flow
+        token.mint(address(crossChainManager), 1000 ether); // Mint tokens to processor
+        vm.prank(address(crossChainManager));
+        token.approve(address(crossChainManager), type(uint).max); // Processor approves bridge logic
+    }
+
+    function testInit() public override(ModuleTest) {
+        assertEq(
+            address(paymentProcessor.orchestrator()), address(_orchestrator)
+        );
+    }
+
+    function testReinitFails() public override(ModuleTest) {
+        vm.expectRevert(OZErrors.Initializable__InvalidInitialization);
+        paymentProcessor.init(_orchestrator, _METADATA, abi.encode(1));
+    }
+
+    function test_ProcessPayments_singlePayment() public {
+        _setupSinglePayment(recipient, 100 ether);
+
+        // Get the client interface
+        IERC20PaymentClientBase_v1 client =
+            IERC20PaymentClientBase_v1(address(paymentClient));
+
+        // Expect the event
+        vm.expectEmit(true, true, true, true);
+        emit PaymentProcessed(
+            0, // paymentId
+            recipient,
+            address(token),
+            100 ether
+        );
+
+        // Process payments
+        crossChainManager.processPayments(client, executionData);
+    }
+
+    function test_ProcessPayments_multiplePayment() public {
+        // Setup mock payment orders that will be returned by the mock
+        address[] memory setupRecipients = new address[](3);
+        setupRecipients[0] = address(0x1);
+        setupRecipients[1] = address(0x2);
+        setupRecipients[2] = address(0x3);
+        uint[] memory setupAmounts = new uint[](3);
+        setupAmounts[0] = 100 ether;
```

Comment:
Working on this !

cc: @leeftk

Reply:
Pushed the changes to the test branch
cc: @leeftk

Reply:
@zzzuhaibmohd we should be using bounds for the upper and lower limits instead of assume, vm.assume discards runs while bound constrains them, this is better for constraining variables. Lets update this please.